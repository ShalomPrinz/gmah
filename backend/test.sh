#!/bin/bash

if [[ $1 == "--coverage" ]]; then
    coverage run -m unittest discover tests
    coverage html
else
    python -m unittest discover tests
fi
