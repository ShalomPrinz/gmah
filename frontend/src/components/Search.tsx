interface SearchProps {
  onChange: (query: string) => void;
}

const Search = ({ onChange }: SearchProps) => (
  <input
    className="form-control my-3 fs-1 p-4"
    onChange={(e) => onChange(e.currentTarget.value)}
    placeholder="Search..."
    type="text"
  />
);

export default Search;
