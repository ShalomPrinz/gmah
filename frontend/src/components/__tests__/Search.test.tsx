import { describe, expect, it, vi } from "vitest";

import { render, screen, setupUser } from "../../../test";

import Search from "../Search";

describe("Search", () => {
  describe("snapshot", () => {
    it("should render empty search box", () => {
      const { asFragment } = render(
        <Search onChange={() => {}} placeholder="" />
      );
      expect(asFragment()).toMatchSnapshot();
    });

    it("should render search box with a placeholder", () => {
      const { asFragment } = render(
        <Search onChange={() => {}} placeholder="some text" />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("behavior", () => {
    it("should get typed text when user types", async () => {
      const user = setupUser();
      const onChange = vi.fn((input: string) => {});
      render(<Search onChange={onChange} placeholder="" />);

      const element = screen.getByRole("searchbox");
      const text = "hello";
      await user.type(element, text);

      expect(onChange).toBeCalledWith(text);
      expect(onChange).toBeCalledTimes(text.length);
    });
  });
});
