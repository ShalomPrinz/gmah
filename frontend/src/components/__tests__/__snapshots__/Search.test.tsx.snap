// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search > snapshot > should render empty search box 1`] = `
<DocumentFragment>
  <input
    class="form-control search-input my-3 fs-1 p-4"
    placeholder=""
    style="border: 5px solid #a4d2f5;"
    type="search"
  />
</DocumentFragment>
`;

exports[`Search > snapshot > should render search box with a placeholder 1`] = `
<DocumentFragment>
  <input
    class="form-control search-input my-3 fs-1 p-4"
    placeholder="some text"
    style="border: 5px solid #a4d2f5;"
    type="search"
  />
</DocumentFragment>
`;
